Amplify/Categories/DataStore/Model/Schema/ModelSchema+Definition.swift

```diff
@@ -7,36 +7,22 @@
 
 import Foundation
 
-// TODO: Add the rest of the AppSync scalar types
-// https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html
-public enum ModelFieldType: CustomStringConvertible {
+/// Defines the type of a `Model` field.
+public enum ModelFieldType {
 
     case string
     case int
     case double
     case date
     case dateTime
     case time
+    case timestamp
     case bool
-    case `enum`(Any.Type)
+    case `enum`(type: ModelEnum.Type)
+    case type(_ type: Codable.Type)
```

Comment:
Maybe `customType` or something a bit more descriptive? Do we need to differentiate that this is a `nonModel` (or something that actually reads better in English)?

Reply:
I like `customType`. The other options I discussed with the JS team were:
- `embeddedType`
- `nestedType`

"non-model" is how we've been referring to it internally due to the lack os agreement on the naming :D

Thoughts on `.customType` vs `.embeddedType` vs `.nestedType`?

Reply:
"embedded" & "nested" have implications that their definitions are somehow self-contained within the parent type. I'd favor "customType", which would stand in opposition to "modelType".